src/shared/varlink.c

```diff
@@ -2234,6 +2319,9 @@ static int connect_callback(sd_event_source *source, int fd, uint32_t revents, v
                 return varlink_server_log_errno(ss->server, errno, "Failed to accept incoming socket: %m");
         }
 
+        if (setsockopt_int(cfd, SOL_SOCKET, SO_PASSCRED, true) < 0)
```

Comment:
setsockopt_int() is our helper call, and all our functions return errors as negative return values, not via `errno`. Hence please change to:

```c
r = setsockopt_int(…);
if (r < 0)
        return varlink_server_log_errno(ss->server, r, …
```